Mathlib/Topology/ContinuousFunction/CompactlySupported.lean

```diff
@@ -0,0 +1,540 @@
+/-
+Copyright (c) 2024 Yoh Tanimoto. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yoh Tanimoto
+-/
+import Mathlib.Topology.ContinuousFunction.CocompactMap
+import Mathlib.Topology.ContinuousFunction.ZeroAtInfty
+import Mathlib.Topology.Support
+
+/-!
+# Compactly supported continuous functions
+
+In this file, we define the type `C_c(α, β)` of compactly supported continuous functions and the
+class `CompactlySupportedContinuousMapClass`, and prove basic properties.
+
+## Main definitions and results
+
+This file contains various instances such as `Add`, `Mul`, `SMul F C_c(α, β)` when `F` is a class of
+continuous functions.
+When `β` has more structures, `C_c(α, β)` inherits such structures as `AddCommGroup`,
+`NonUnitalRing`  and `StarRing`.
+
+When the domain `α` is compact, `ContinuousMap.liftCompactlySupported` gives the identification
+`C(α, β) ≃ C_c(α, β)`.
+
+## TODO
+
+* Create more intances of algebraic structures (e.g., `NonUnitalSemiring`) once the necessary
+  type classes (e.g., `TopologicalRing`) are sufficiently generalized.
```

Comment:
I think you copied this from another file and it's no longer applicable.

Reply:
removed.